### PR TITLE
Support multiple accepted-public-api.json files

### PR DESCRIPTION
**Pull request by asodja**
So we can avoid having conflicts with main branch when we will migrate to Provider API.

We now discover `.json` files in `src/changes/accepted-changes` directory.

Fixes https://github.com/gradle/gradle/issues/26573

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle [hunk 1]
```diff
@@ -47,8 +47,15 @@ repositories {
     }
 }
 
-def apiChangesJsonFile = layout.projectDirectory.file("src/changes/accepted-public-api-changes.json")
-def acceptedViolations = AcceptedApiChanges.parse(providers.fileContents(apiChangesJsonFile).asText.get())
+// We support multiple accepted public api changes JSON files, but "accepted-public-api-changes.json"
+// is the "main" one that is also shown in HTML report
+def apiChangesJsonDirectory = layout.projectDirectory.dir("src/changes/accepted-changes")
```
Comment (asodja) [inline build-logic/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle:52]: 💭 We now have `src/changes/accepted-changes` where any `.json` file is discovered

**Comment by asodja** [comment]
CI failure doesn't look related.

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/CleanAcceptedApiChanges.groovy [hunk 2]
```diff
@@ -27,15 +28,17 @@ import org.gradle.work.DisableCachingByDefault
...
-    File jsonFile
+    @InputDirectory
+    abstract DirectoryProperty getJsonFileDirectory()
 
     @TaskAction
     void clean() {
-        AcceptedApiChangesJsonFileManager jsonFileManager = new AcceptedApiChangesJsonFileManager()
-        jsonFileManager.emptyAcceptedApiChanges(jsonFile)
+        jsonFileDirectory.asFile.get().listFiles().each {
+            AcceptedApiChangesJsonFileManager jsonFileManager = new AcceptedApiChangesJsonFileManager()
```
Comment (lptr) [inline build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/CleanAcceptedApiChanges.groovy:40]: 💅 Can we reuse the same instance?

**Review by lptr — APPROVED** [review]
LGTM. As so often happens, it took quite a bit more changes than I'd have expected initially.

File: build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/CleanAcceptedApiChanges.groovy (hunk 2 above)
Comment (asodja) [reply build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/CleanAcceptedApiChanges.groovy:40]: Good catch, I will fix it

**Comment by ljacomet** [comment]
This most likely has an impact on promotion jobs, in particular branching.
This change should be coordinated with changes in the promotion job or at least a blocking issue needs to be raised for promotion.

**Comment by ljacomet** [comment]
I believe that's it, thanks for doing a PR already.

**Pull request merged**